cranelift-filetests/src/function_runner.rs

```diff
@@ -1,53 +1,280 @@
+}
+
+impl Display for Operator {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        f.write_str(match self {
+            Self::EQ => "==",
+            Self::NEQ => "!=",
+        })
+    }
+}
+
+impl FromStr for Operator {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "==" => Ok(Self::EQ),
+            "!=" => Ok(Self::NEQ),
+            _ => Err(format!("Unsupported operator {}", s)),
+        }
+    }
+}
+
+macro_rules! try_into_int {
+    ($type: ty, $name: ident) => {
+        struct $name {}
+        impl $name {
+            fn try_into(d: ConstantData) -> Result<$type, String> {
+                if d.len() == (mem::size_of::<$type>()) {
+                    let v = d.into_vec().into_iter().rev();
+                    let mut r: $type = 0;
+                    for b in v {
+                        let (mut shifted, _) = r.overflowing_shl(8);
+                        shifted |= b as $type;
+                        r = shifted;
+                    }
+                    Ok(r)
+                } else {
+                    Err(format!(
+                        "Incorrect vector size: {}, expected {}",
+                        d.len(),
+                        mem::size_of::<$type>()
+                    ))
+                }
+            }
+        }
+    };
+}
+
+try_into_int!(u16, ConstantDataU16);
+try_into_int!(u32, ConstantDataU32);
+try_into_int!(u64, ConstantDataU64);
+try_into_int!(u128, ConstantDataU128);
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum Value {
+    Bool(bool),
+    F32(f32),
+    F64(f64),
+    I8(u8),
+    I16(u16),
+    I32(u32),
+    I64(u64),
+    I128(u128),
+}
+
+impl Value {
+    fn parse_as(s: &str, ty: types::Type) -> Result<Self, String> {
+        if ty.is_vector() {
+            return Err(format!("Expected a non-vector type, not {}", ty));
+        }
+
+        let mut p = Parser::new(s);
+
+        let constant_data = match ty {
+            types::I8 => {
+                let data = p.match_uimm8("Expected a 8-bit unsigned integer").unwrap();
+                Self::I8(data.into())
+            }
+            types::I16 => {
+                let data = p.match_constant_data(ty).unwrap();
+                Self::I16(ConstantDataU16::try_into(data).unwrap())
+            }
+            types::I32 => {
+                let data = p
+                    .match_uimm32("Expected a 32-bit unsigned integer")
+                    .unwrap();
+                Self::I32(data.into())
+            }
+            types::I64 => {
+                let data = p
+                    .match_uimm64("Expected a 64-bit unsigned integer")
+                    .unwrap();
+                Self::I64(data.into())
+            }
+            types::I128 => {
+                let data = p.match_constant_data(ty).unwrap();
+                let data = ConstantDataU128::try_into(data).unwrap();
+                Self::I128(data.rotate_left(64))
```

Comment:
???

Reply:
Swaps left and right qwords, if causes more problems than it solves, I can remove I128 support from the PR.

Reply:
`ConstantData` is stored in little endian format. `ConstantDataU128` converts it to a native endian `u128`, so no swap should be necessary.

Reply:
It isn't clear to me what the resolution is here, if any.